src/main/kotlin/org/rust/utils/io.kt

```diff
@@ -0,0 +1,29 @@
+package org.rust.utils
+
+import com.intellij.util.io.DataExternalizer
+import java.io.DataInput
+import java.io.DataOutput
+
+interface RustDataExternalizer<T> : DataExternalizer<T> {
+    override fun save(output: DataOutput, value: T)
+    override fun read(input: DataInput): T
```

Comment:
I don't get what these overrides do. Couldn't we just provide `nullable` as extension method for `DataExternalizer` itself?